File: im-desktop/src/types/scheduled-message.ts
/**
 * 定时消息类型定义
 */

export type ScheduledMessageStatus = 'PENDING' | 'SENT' | 'CANCELLED' | 'FAILED';

export interface ScheduledMessage {
  id: number;
  senderId: number;
  receiverId: number;
  content: string;
  status: ScheduledMessageStatus;
  scheduledTime: string;
  sentTime?: string;
  failureReason?: string;
  createdAt: string;
  receiverNickname?: string;
  receiverAvatar?: string;
}

export interface CreateScheduledMessageRequest {
  receiverId: number;
  content: string;
  scheduledTime: string;
}

export interface UpdateScheduledMessageRequest {
  receiverId?: number;
  content?: string;
  scheduledTime?: string;
}

export interface ScheduledMessageStats {
  pendingCount: number;
}

export const StatusLabel: Record<ScheduledMessageStatus, string> = {
  PENDING: '待发送',
  SENT: '已发送',
  CANCELLED: '已取消',
  FAILED: '发送失败',
};

export const StatusColor: Record<ScheduledMessageStatus, string> = {
  PENDING: 'blue',
  SENT: 'green',
  CANCELLED: 'default',
  FAILED: 'red',
};

File: im-backend/src/main/java/com/im/backend/model/ScheduledMessage.java
package com.im.backend.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import java.time.LocalDateTime;

/**
 * 定时消息实体类
 * 用于存储用户预设的定时发送消息
 */
@Entity
@Table(name = "scheduled_messages", indexes = {
    @Index(name = "idx_sender_id", columnList = "senderId"),
    @Index(name = "idx_receiver_id", columnList = "receiverId"),
    @Index(name = "idx_status", columnList = "status"),
    @Index(name = "idx_scheduled_time", columnList = "scheduledTime")
})
public class ScheduledMessage {

    public enum Status {
        PENDING,      // 待发送
        SENT,         // 已发送
        CANCELLED,    // 已取消
        FAILED        // 发送失败
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Column(nullable = false)
    private Long senderId;

    @NotNull
    @Column(nullable = false)
    private Long receiverId;

    @NotBlank
    @Size(max = 5000)
    @Column(nullable = false, length = 5000)
    private String content;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Status status = Status.PENDING;

    @NotNull
    @Column(nullable = false)
    private LocalDateTime scheduledTime;

    @Column
    private LocalDateTime sentTime;

    @Column
    private String failureReason;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Long getSenderId() { return senderId; }
    public void setSenderId(Long senderId) { this.senderId = senderId; }

    public Long getReceiverId() { return receiverId; }
    public void setReceiverId(Long receiverId) { this.receiverId = receiverId; }

    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }

    public Status getStatus() { return status; }
    public void setStatus(Status status) { this.status = status; }

    public LocalDateTime getScheduledTime() { return scheduledTime; }
    public void setScheduledTime(LocalDateTime scheduledTime) { this.scheduledTime = scheduledTime; }

    public LocalDateTime getSentTime() { return sentTime; }
    public void setSentTime(LocalDateTime sentTime) { this.sentTime = sentTime; }

    public String getFailureReason() { return failureReason; }
    public void setFailureReason(String failureReason) { this.failureReason = failureReason; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }
}
